src/components/RegionSelector/index.tsx

```diff
@@ -125,24 +129,27 @@ const RegionSelector: React.FC<RegionSelectorProps> = ({
                             active
                               ? 'text-white bg-indigo-600'
                               : 'text-gray-300'
-                          } cursor-default select-none relative py-2 pl-8 pr-4`}
+                          } cursor-default select-none relative py-2 pl-8 pr-4 flex items-center`}
                         >
+                          {currentSettings.region && (
+                            <span className="mr-2 text-lg">
+                              {countryCodeEmoji(currentSettings.region)}
+                            </span>
+                          )}
                           <span
                             className={`${
                               selected ? 'font-semibold' : 'font-normal'
                             } block truncate`}
                           >
                             {intl.formatMessage(messages.regionServerDefault, {
-                              applicationTitle:
-                                currentSettings.applicationTitle,
                               region: currentSettings.region
                                 ? intl.formatDisplayName(
                                     currentSettings.region,
                                     {
                                       type: 'region',
                                       fallback: 'none',
                                     }
-                                  ) ?? currentSettings.region
```

Comment:
You still need the `currentSettings.region` for the fallback when formatDisplayName cant find the localization of the region.